experimental/packages/otlp-grpc-exporter-base/src/internal-types.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type * as protobuf from 'protobufjs';
+
+export interface ExportType<T, R = T & { toJSON: () => unknown }> {
+  encode(message: T, writer?: protobuf.Writer): protobuf.Writer;
+  decode(reader: protobuf.Reader | Uint8Array, length?: number): R;
+}
```

Comment:
Is this the only place protobufjs peer dependency is needed? Might be possible to use an interface or a type we hand-roll here to avoid the peer dep?

Reply:
it's used in the code that is generated by `protobufjs-cli` when running the `protos` script of this package, so we cannot avoid the dependency there unfortunately. However, `@gprc/proto-loader` depended on `protobufjs` already, so we're still reducing our dependency count by cutting out `@gprc/proto-loader`. :slightly_smiling_face: